src/cache_hash.cr

```diff
@@ -24,19 +24,18 @@ class CacheHash(V)
     end
   end
 
-  private def delete(key : String) : String | Nil
+  private def delete(key : String) : Nil
     @time_hash.delete key
     @kv_hash.delete key
     nil
   end
 
   def purge_stale
-    @kv_hash.select! do |k, v|
-      if cached_time = @time_hash[k]?
-        cached_time > Time.now - @cache_time_span
+    @kv_hash.each_key do |key|
+      if cached_time = @time_hash[key]?
+        delete key unless cached_time > Time.now - @cache_time_span
```

Comment:
The reason I didn't do something like this i because I was concerned that deleting an element from the hash during the loop could possibly cause the loop to skip over an element. Is that not an issue here?

But thanks for noticing that its not deleting from the time hash.

Reply:
This is not an issue. It won't skip over an element, in crystal a `Hash` is basically a linked list of `Entry(K, V)`. If you delete the current entry it is only removed from the chain, but the current entry still points to the next one. The `#each` method which `#each_key` uses is [here](https://github.com/crystal-lang/crystal/blob/e2a1389e8165fb097c785524337d7bb7b550a086/src/hash.cr#L305).

Reply:
awesome. didn't know that.